packages/documentation/.storybook/addons/styles-switcher/StylesSwitcher.tsx

```diff
@@ -94,6 +96,49 @@ function StylesSwitcher() {
     );
   }, [preview, currentTheme, currentChannel]);
 
+  /**
+   * Sets the design system styles import SCSS file to the correct theme and channel file
+   */
+  useEffect(() => {
+    if (!stylesCodeBlocks) return;
+
+    const t = currentTheme.toLowerCase();
+    const c = currentChannel.toLowerCase();
+    const domain = "'@swisspost/design-system-styles/";
+
+    stylesCodeBlocks.forEach(stylesCodeBlock => {
+      let sourceArray = stylesCodeBlock.querySelectorAll('.token.string');
+      if (sourceArray.length) {
+        sourceArray.forEach((s, i) => {
+          let source = s.innerHTML;
+          // Remove the domain from the source to make sure we don't override it
+          source = source.replace(domain, '');
+
+          // Check if one of the themes or channels are in the scss path
+          let theme = THEMES.find(t => source.indexOf(t.toLowerCase()) > -1);
+          let channel = CHANNELS.find(c => source.indexOf(c.toLowerCase()) > -1);
+
+          const updateTheme = theme && theme.toLowerCase() !== t;
+          const updateChannel = channel && channel.toLowerCase() !== c;
+
+          // Only change the source if theme or channel needs to be changed
+          if (source && (updateTheme || updateChannel)) {
+            if (updateTheme) {
+              source = source.replace((theme as string).toLowerCase(), t);
+            }
+
+            if (updateChannel) {
+              source = source.replace((channel as string).toLowerCase(), c);
+            }
+
+            const newSource = domain + source;
+            sourceArray[i].innerHTML = newSource;
```

Comment:
Isn't `sourceArray[i].innerHTML` the same as `source`?

Reply:
yes, pretty much but if I put source = newSource, that only updates the source variable and not the actual innerHTML of my element.